config/vite.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+use craft\helpers\App;
+
+
+// Decides which port to use for devServerPublic. This allows
+// you to visit the http or https port. If you plant to only
+// use http, make sure that PRIMARY_SITE_URL is set to http
+//
+// Match ports to .ddev/config.yaml -> web_extra_exposed_ports
+$host = Craft::$app->getRequest()->getIsConsoleRequest()
+    ? App::env('PRIMARY_SITE_URL')
+    : Craft::$app->getRequest()->getHostInfo();
+$httpPort = 3000;
+$httpsPort = 3001;
+$devServerPort = str_starts_with($host, 'https') ? $httpsPort : $httpPort;
```

Comment:
For a while I was having difficulty getting `https` working with DDEV (I've got it working great now). 

Figured it wasn't too much extra code to accommodate other's who might hit that issue.